LoopBack/LBModel.m

```diff
@@ -122,9 +110,21 @@ - (LBModel *)modelWithDictionary:(NSDictionary *)dictionary {
             }
 
             const char *type = property_getAttributes(property);
-            // if the property type is NSDate, convert the string to a date object
-            if ([obj isKindOfClass:[NSString class]] && strncmp(type, "T@\"NSDate\",", 11) == 0) {
-                obj = [jsonDateFormatter dateFromString:obj];
+            if ([obj isKindOfClass:[NSString class]]) {
+                // if the property type is NSDate, convert the string to a date object
+                if (strncmp(type, "T@\"NSDate\",", 11) == 0) {
+                    obj = [SLObject dateFromEncodedProperty:obj];
+                }
+            } else if ([obj isKindOfClass:[NSDictionary class]]) {
+                // if the property type is NSMutableData, convert the json object to a data object
+                if (strncmp(type, "T@\"NSMutableData\",", 18) == 0 ||
+                    strncmp(type, "T@\"NSData\",", 11) == 0) {
+                    obj = [SLObject dataFromEncodedProperty:obj];
+                }
+                // if the property type is CLLocation, convert the json object to a location object
+                else if (strncmp(type, "T@\"CLLocation\",", 15) == 0) {
+                    obj = [SLObject locationFromEncodedProperty:obj];
+                }
```

Comment:
The parameter conversions back to iOS data types are needed to be performed here in `LBModel` since it requires access to model property type attributes.